examples/react/modules/lobby/components/lobby.js

```diff
@@ -0,0 +1,339 @@
+            onClick={() => this.onClickInstance(i)}
+          >{`[${inst.game_id}] ${inst.players.length} players`}</td>
+        </tr>
+      );
+    }
+
+    const getPhaseVisibility = phase => {
+      return this.state.phase !== phase ? 'hidden' : 'phase';
+    };
+    return (
+      <div id="lobby-view" style={{ padding: 50 }}>
+        <h1>Lobby</h1>
+        <div id="phase-enter" className={getPhaseVisibility('enter')}>
+          <p className="phase-title">Choose a player name:</p>
+          <input
+            type="Text"
+            value={this.state.playerName}
+            onChange={evt => this.onChangePlayerName(evt)}
+            onKeyPress={evt => this.onKeyPress(evt)}
+          />
+          <span className="buttons">
+            <button className="buttons" onClick={() => this.onClickEnter()}>
+              Enter
+            </button>
+          </span>
+          <br />
+          <span id="name-error" className="error-msg">
+            {this.state.nameErrorMsg}
+            <br />
+          </span>
+        </div>
+        <div id="phase-join" className={getPhaseVisibility('join')}>
+          <div id="games">
+            <p className="phase-title">Select a game:</p>
+            <table>
+              <tbody>{game_rows}</tbody>
+            </table>
+            <span id="games-error" className="error-msg">
+              {this.state.gamesErrorMsg}
+              <br />
+            </span>
+          </div>
+          <div id="instances">
+            <p className="phase-title">Select or create an instance:</p>
+            <div id="instances-table">
+              <table>
+                <tbody>{inst_rows}</tbody>
+              </table>
+            </div>
+            <div className="buttons">
+              <button onClick={() => this.onClickJoin()}>Join</button>
+              <button onClick={() => this.onClickCreate()}>Create</button>
+              <button onClick={() => this.onClickExitLobby()}>Exit</button>
+            </div>
+            <span id="instances-error" className="error-msg">
+              {this.state.instancesErrorMsg}
+              <br />
+            </span>
+          </div>
+        </div>
+        <div id="phase-play" className={getPhaseVisibility('play')}>
+          {this.state.game ? (
+            <this.state.game.app
+              gameID={this.state.game.gameID}
+              playerID={this.state.game.playerID}
+              credentials={this.state.game.playerCredentials}
+            />
+          ) : (
+            ''
+          )}
+          <div className="buttons">
+            <button onClick={() => this.onClickExitGame()}>Exit</button>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  onKeyPress(event) {
+    if (this.state.phase === 'enter' && event.key === 'Enter') {
+      this.onClickEnter();
+    }
+  }
+
+  onChangePlayerName(event) {
+    const name = event.target.value.trim();
+    this.setState({
+      ...this.state,
```

Comment:
per the react docs `setState` performs a shallow merge into the new state, so it is not needed to pass `...this.state` , it is redundant

Reply:
Thanks for the tip, fixed in commit d183131